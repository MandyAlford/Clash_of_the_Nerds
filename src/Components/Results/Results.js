import React from 'react';
import { connect } from 'react-redux';
import TriviaResults from '../TriviaResults/TriviaResults';
import './Results.scss';

const Results = (props) => {
  const allWrong = props.incorrectGuess.map(guess => {
    console.log(guess);
    return <TriviaResults result={guess} />
  })
  return (
    <section id='all-results'>
      {allWrong}
    </section>
  )
}

export const mapStateToProps = state => ({
  correctGuess: state.correctGuess,
  incorrectGuess: state.incorrectGuess
});

export default connect(mapStateToProps)(Results)
