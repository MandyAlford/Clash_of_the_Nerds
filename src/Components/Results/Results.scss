#all-results {
  overflow-y: scroll;
  height: 100vh;
}

.total-results {
  position: fixed;
  background-color: rgba(255, 255, 255, 0.9);
  border: 5px solid #662324;
  height: 200px;
  width: 400px;
  padding: 55px;
  z-index: 100;
  float: left;
  margin: 100px;
  font-size: 30px;
}

.wrong-results {
  position: relative;
  z-index: 100;
  float: right;
  margin-right: 10%;
  margin-top: 100px;
  margin-bottom: 300px;
}

#percent-correct {
  margin-top: 350px;
}

.wrong-title {
  font-size: 30px;
  color: #FFF;
  margin-bottom: 20px;
  margin-top: -55px;
}

.correct-results {
  position: relative;
  z-index: 100;
  float: right;
  margin-top: 100px;
  margin-bottom: 300px;
  margin-right: 50px;
  margin-bottom: 300px;

}
