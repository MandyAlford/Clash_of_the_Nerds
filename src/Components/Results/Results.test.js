import React from 'react';
import Results from './Results';
import { shallow } from 'enzyme';

describe('Results', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Results
      incorrectGuess={[]}
      correctGuess={[]}
    />)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
