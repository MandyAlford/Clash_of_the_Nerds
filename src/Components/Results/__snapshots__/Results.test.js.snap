// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results should match snapshot 1`] = `
<section>
  <div
    className="total-results"
  >
    <h1>
      You Scored 
      20
       % Correct
    </h1>
  </div>
  <div
    className="total-results"
    id="percent-correct"
  >
    <h1>
      You Missed 
      8
       out of 10
    </h1>
  </div>
  <div
    className="wrong-results"
  >
    <h1
      className="wrong-title"
    >
      You Missed
    </h1>
    <Connect(TriviaResults)
      key="0.1"
      result={
        Object {
          "category": "Mythology",
          "correct_answer": "The Sea",
          "difficulty": "easy",
          "incorrect_answers": Array [
            "War",
            "Sun",
            "Fire",
          ],
          "question": "The greek god Poseidon was the god of what?",
          "type": "multiple",
          "userGuess": "War",
        }
      }
    />
  </div>
  <div
    className="correct-results"
  >
    <h1
      className="wrong-title"
    >
      You Answered Correct
    </h1>
    <Connect(TriviaResults)
      key="0.1"
      result={
        Object {
          "category": "Mythology",
          "correct_answer": "true",
          "difficulty": "easy",
          "incorrect_answers": Array [
            "false",
          ],
          "question": "Loki dressed as a woman in norse mythology",
          "type": "true/false",
          "userGuess": "true",
        }
      }
    />
    <Connect(TriviaResults)
      key="0.1"
      result={
        Object {
          "category": "Mythology",
          "correct_answer": "Athena",
          "difficulty": "medium",
          "incorrect_answers": Array [
            "Ares",
            "Artemis",
            "Apollo",
          ],
          "question": "This Greek mythological figure is the god/goddess of battle strategy (among other things).",
          "type": "multiple",
          "userGuess": "Athena",
        }
      }
    />
  </div>
</section>
`;
